### DISCUSSION TITLE
How do I create a new format to detect?

### DISCUSSION THREAD

**Discussion by Ytrog** [horsicq/Detect-It-Easy#228] [Q&A]
Hello and thanks for the wonderful tool 😊

I have a question about how to create a new detection signature? I see the `.sg` files in the `/db` folder, however I have a few questions as I can't find documentation on the format:

1. What does the `init(…)` function do?
2. What is needed to implement a `detect(bShowType, bShowVersion, bShowOptions)` function?
3. How do you use `Binary.compare("…")`? What is the pattern that goes into the string and what gets returned?

I want to make a detection signature for JPEG-XL files as they are currently just classified under the generic "Binary" class, however I find it hard to get started with the signature language as it seems to be lacking documentation (or I'm really looking in the wrong places).

Thanks in advance 🤓👍

**Comment by horsicq** [comment]
Hello! Thanks a lot for your support!

1) It is just for init some variables https://github.com/horsicq/Detect-It-Easy/blob/master/db/_init#L29. You can just ignore it or use your own function if you need it.

2) The same https://github.com/horsicq/Detect-It-Easy/blob/master/db/result#L6

3) The documentation: https://github.com/horsicq/Detect-It-Easy/blob/master/help/Binary.md

We can add the file format rapidly. Do you need just detect + version ? Or you need some extra information?

**Reply by Ytrog** [reply]
Just detect + version would be nice 😊👍

**Comment by horsicq** [comment]
https://github.com/horsicq/Detect-It-Easy/blob/master/db/Binary/image.JPEGXL.1.sg

**Comment by DosX-dev** [comment]
Hello! Is your problem solved?

**Reply by Ytrog** [reply]
Certainly. Thank you 😁👍
